GEON-PPANG-iOS/Global/Network/Base/AuthPlugin.swift

```diff
@@ -0,0 +1,73 @@
+//
+//  AuthPlugin.swift
+//  GEON-PPANG-iOS
+//
+//  Created by JEONGEUN KIM on 2023/09/04.
+//
+
+import Foundation
+
+import Moya
+
+final class AuthPlugin: PluginType {
+    
+    // Request를 보낼 때 호출
+    func willSend(_ request: RequestType, target: TargetType) {
+        guard let httpRequest = request.request else {
+            print("--> 유효하지 않은 요청")
+            return
+        }
+        let url = httpRequest.description
+        let method = httpRequest.httpMethod ?? "unknown method"
+        var log = "----------------------------------------------------\n\n[\(method)] \(url)\n\n----------------------------------------------------\n"
+        log.append("API: \(target)\n")
+        if let headers = httpRequest.allHTTPHeaderFields, !headers.isEmpty {
+            log.append("header: \(headers)\n")
+        }
+        if let body = httpRequest.httpBody, let bodyString = String(bytes: body, encoding: String.Encoding.utf8) {
+            log.append("\(bodyString)\n")
+        }
+        log.append("------------------- END \(method) --------------------------")
+        print(log)
```

Comment:
하암? ㅋㅋㅋㅋㅋ 근데 이거 좋은 방법 있는게
```suggestion
        #if DEBUG
        var log = "----------------------------------------------------\n\n[\(method)] \(url)\n\n----------------------------------------------------\n"
        log.append("API: \(target)\n")
        if let headers = httpRequest.allHTTPHeaderFields, !headers.isEmpty {
            log.append("header: \(headers)\n")
        }
        if let body = httpRequest.httpBody, let bodyString = String(bytes: body, encoding: String.Encoding.utf8) {
            log.append("\(bodyString)\n")
        }
        log.append("------------------- END \(method) --------------------------")
        print(log)
        #endif
```
이렇게 # 부분 넣어주면 릴리즈 버전에서는 실행 안되고 디버깅 버전에서만 실행됨요 ㅎ

Reply:
넣어보겠움

Reply:
8737c8ef2579c29f740698373fd18a5d5ed0e89e